src/test/groovy/graphql/schema/SchemaTransformerTest.groovy

```diff
@@ -692,4 +692,23 @@ type Query {
         then:
         newFoo.getFieldDefinition("changed") != null
     }
+
+    def "if nothing changes in the schema transformer, we return the same object"() {
+
+        def schema = TestUtil.schema("type Query { f : String }")
+
+        def fieldChanger = new GraphQLTypeVisitorStub() {
+
+            @Override
+            TraversalControl visitGraphQLFieldDefinition(GraphQLFieldDefinition node,
+                                                         TraverserContext<GraphQLSchemaElement> context) {
+                return TraversalControl.CONTINUE
+            }
+        }
+
+        when:
+        def newSchema = SchemaTransformer.transformSchema(schema, fieldChanger)
+        then:
+        newSchema == schema
```

Comment:
I think you want `is` or `===`.
I suspect we made this error quite a lot tbh. 
See https://docs.groovy-lang.org/latest/html/documentation/core-operators.html#_relational_operators